### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by estebv**
-Ejercicio  de https://vulky.atlassian.net/jira/software/c/projects/BACK/boards/7?modal=detail&selectedIssue=BACK-7

Este es un ejercicio de POO desarrollado en Java.

-Herencia
-Polimorfismo
-Interfaz
-Objetos como parámetro
-clase de modelo
-Encapsulación

-------------------------------------

-exercise  from https://vulky.atlassian.net/jira/software/c/projects/BACK/boards/7?modal=detail&selectedIssue=BACK-7

This is an OOP exercise developed in Java.

-Inheritance
-polymorphism
-Interface
-Objects as parameter
-model class
-encapsulation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Compiled class file
```
Comment (MauroDecodificado) [inline .gitignore:1]: Nice 👏

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/AccionPersona.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+package com.mycompany.ejercicio1vuky;
+
+
+public interface AccionPersona {
+    
+    public void caminar();
+    public void saltar();
+    public void gritar();
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/AccionPersona.java]: Please delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.mycompany.ejercicio1vuky;
+
+public class Main {
+
+    public static void main(String[] args) {
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java:6]: Delete empty space

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+
+package com.mycompany.ejercicio1vuky;
+
+
+public class ModeloPersona{
+    
+ private String colorP;
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: What is colorP?

Please rename to a more clear naming variable

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ private  String nombre;
+ private int edad;
+
+    public String getColorP() {
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: Update method name to reflect the same than attribute

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        return colorP;
+    }
+
+    public void setColorP(String colorP) {
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: Same here, Update method name to reflect the same than attribute

File: README.md [hunk 7]
```diff
@@ -1 +1,15 @@
-# Ejerciciospoo
\ No newline at end of file
+# EjerciciosPoo
+
+Es un ejercicio donde se tocan los siguientes conceptos:
+
+- herencia(implementacion y extencion)
```
Comment (MauroDecodificado) [inline README.md]: Please check accents

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public int getEdad() {
+        return edad;
+    }
+
+    public void setEdad(int edad) {
+        this.edad = edad;
+    }
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: Delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                + pedrito.informacionGlobal.getEdad() + "años");
+
+        Pepita pepota = new Pepita();
+        ModeloPersona informacionPepita = new ModeloPersona();
+        informacionPepita.setColorP("rubia");
+        informacionPepita.setNombre("susana");
+        informacionPepita.setEdad(28);
+        pepota.llenarInformacion(informacionPedro);
+        pepota.caminar();
+        pepota.tocarFlauta("");
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java]: Please sent something into the ""

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java]: Delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java [hunk 11]
```diff
@@ -0,0 +1,17 @@
+
+package com.mycompany.ejercicio1vuky;
+
+
+public class Pedro extends Persona{
+    
+    
+    
+    public void tocarPiano(String cancion){
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: Please do some functionality here, you can maybe print a message with the cancion you have received

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+public class Pedro extends Persona{
+    
+    
+    
+    public void tocarPiano(String cancion){
+       
+        
+    }
+    
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: Delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+package com.mycompany.ejercicio1vuky;
+
+
+public class Pedro extends Persona{
+    
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: Delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java [hunk 14]
```diff
@@ -0,0 +1,14 @@
+
+package com.mycompany.ejercicio1vuky;
+
+
+public class Pepita extends Persona{
+    
+    public void tocarFlauta( String cancion){
+        System.out.println("toca"+cancion);
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java]: Awesome you can implement the same in Pedro class

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+
+package com.mycompany.ejercicio1vuky;
+
+
+public class Pepita extends Persona{
+    
+    public void tocarFlauta( String cancion){
+        System.out.println("toca"+cancion);            
+        
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java]: Delete empty spaces

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
+package com.mycompany.ejercicio1vuky;
+
+public class Persona implements AccionPersona {
+
+    public ModeloPersona informacionGlobal;
+
+    public void llenarInformacion(ModeloPersona informacionlocal) {
+        informacionGlobal = informacionlocal;
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java]: Delete empty space

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public void caminar() {
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:12]: Please add a functionality in the method

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public void saltar() {
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:17]: Please add a functionality in the method

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public void gritar() {
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:22]: Please add a functionality in the method

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+    }
+}
+
+
```
Comment (MauroDecodificado) [inline ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:28]: Delete those empty spaces

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Check comments please

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java (hunk 3 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java:6]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java (hunk 4 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java (hunk 5 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java (hunk 6 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: ready

File: README.md (hunk 7 above)
Comment (estebv) [reply README.md]: como asi?

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java (hunk 8 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/ModeloPersona.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java (hunk 9 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java (hunk 10 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Main.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java (hunk 11 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java (hunk 12 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: ready

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java (hunk 13 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pedro.java]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java (hunk 14 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java]: ok

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java (hunk 15 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Pepita.java]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java (hunk 20 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:28]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java (hunk 19 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:22]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java (hunk 18 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:17]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java (hunk 17 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java:12]: ready.

File: ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java (hunk 16 above)
Comment (estebv) [reply ejercicio1vulky/src/main/java/com/mycompany/ejercicio1vuky/Persona.java]: ready.

File: README.md (hunk 7 above)
Comment (MauroDecodificado) [reply README.md]: Please in English, I'm referring to implementacion to implementación, please check all of them

File: README.md [hunk 21]
```diff
@@ -1 +1,15 @@
-# Ejerciciospoo
\ No newline at end of file
+# EjerciciosPoo
+
+Es un ejercicio donde se tocan los siguientes conceptos:
```
Comment (MauroDecodificado) [inline README.md:3]: Please add a section for english version

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Please check comments

**Review by MauroDecodificado — APPROVED** [review]
Approved and can be merged

**Pull request merged**